clang/test/Analysis/pointer-sub-notes.c

```diff
@@ -32,3 +32,20 @@ void different_2() {
   int d = p2 - p1; // expected-warning{{Subtraction of two pointers that do not point into the same array is undefined behavior}} \
                    // expected-note{{Subtraction of two pointers that do not point into the same array is undefined behavior}}
 }
+
+int different_3() {
+  struct {
+    int array[5];
+  } a, b;
+  return &a.array[3] - &b.array[2]; // expected-warning{{Subtraction of two pointers that do not point into the same array is undefined behavior}} \
+                                    // expected-note{{Subtraction of two pointers that do not point into the same array is undefined behavior}}
+}
+
+int different_5() {
```

Comment:
```suggestion
int different_4() {
```
Very minor nitpick: `different_3` was followed by `different_5`.